hunt/app/controllers/animals_controller.rb

```diff
@@ -17,27 +18,37 @@ def create
     if @animal.save
       redirect_to animals_path
     else
-      render :new
+      render action: 'new'
     end
   end
 
   def edit
-    @animal = Animal.find(params[:id])
   end
 
   def update
-    @animal = Animal.find(params[:id])
+    @animal.update(animal_params)
 
-    if @animal.update(animal_params)
-      redirect_to animal_path(@animal)
+    if @animal.save
```

Comment:
`update` will actually run validations also, so this could be shortened to:
```
  def update
    if @animal.update(animal_params)
      redirect_to animals_path(@animal)
    else
      render action: 'edit'
    end
  end
```